modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultGenerator.java

```diff
@@ -487,7 +487,26 @@ private Model getParent(Model model) {
                 // TODO revise below as we've already performed unaliasing so that the isAlias check may be removed
                 Map<String, Object> modelTemplate = (Map<String, Object>) ((List<Object>) models.get("models")).get(0);
                 // Special handling of aliases only applies to Java
-                if (modelTemplate != null && modelTemplate.containsKey("model")) {
+                Set<String> javaGenerators = new HashSet<String>(Arrays.asList(
+                        "java",
+                        "jaxrs-cxf-client",
+                        "jaxrs-cxf-extended",
+                        "jaxrs-cxf",
+                        "java-inflector",
+                        "jaxrs-cxf-cdi",
+                        "jaxrs-spec",
+                        "jaxrs-jersey",
+                        "java-msf4j",
+                        "java-pkmst",
+                        "java-play-framework",
+                        "jaxrs-resteasy-eap",
+                        "jaxrs-resteasy",
+                        "java-undertow-server",
+                        "java-vertx",
+                        "spring",
+                        "groovy"
+                ));
+                if (modelTemplate != null && modelTemplate.containsKey("model") &&  javaGenerators.contains(config.getName())) {
```

Comment:
@wing328 what do you think about how I did this?
A more elegant way may be to add a constant in the generators that use isAlias, something like
config.uses_alias
Then in this conditional we would check 
`(modelTemplate != null && modelTemplate.containsKey("model") &&  !config.uses_alias) {`
This search: https://github.com/OpenAPITools/openapi-generator/search?p=1&q=isAlias&unscoped_q=isAlias
shows that not many generators use isAlias.

Reply:
As you mentioned above, I , too, think it's better to add something in "the generators" instead of to make `DefaultGenerator` has a variable indicates which generator needs the special handling.

* How about an instance method like `config.shouldGenerateAliasModels()`?
  * Adding the method into `CodegenConfig` interface could guarantee the behavior.

Reply:
Updated the code to use the existing command line argument generate alias as moddls. This PR sets that boolean to always be on in the Python generator. However the alias model is only generated if it contains validations ir enums.